src/types/form/SearchAdvancedFiltersForm.ts

```diff
@@ -0,0 +1,27 @@
+import type DeepValueOf from '@src/types/utils/DeepValueOf';
+import type Form from './Form';
+
+const INPUT_IDS = {
+    TYPE: 'type',
+    DATE: {
+        AFTER: 'dateAfter',
+        BEFORE: 'dateBefore',
+    },
+} as const;
+
+type InputID = DeepValueOf<typeof INPUT_IDS>;
```

Comment:
I feel this types are unnecessary complicated.
If what gets saved in onyx is a flat object like this:
```
{
   dateBefore
   dateAfter
}
```

then please define INPUT_IDS in the same flat way, so that people don't get confused. Then you can just use `ValueOf` instead of `DeepValueOf`

Reply:
This is possible to save both dates separately in 'dateBefore' and 'dateAfter' keys, but I thought that since both dates are picked in a single form, they should be nested within 'date' object, one level deeper than typical filter keys: 
```
{
   filter1
   date : {
        after
        before
    }
    filter3
    filter4
}
```

Reply:
Yeah I understand the logic, but I dislike that the type is flat but the const object is nested. IMO it would be simpler to work with if both would be the same.